File: trail2/요일 맞추기/guess-day-of-week.java
import java.util.Scanner;

public class Main {
    static int[] mm = {0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    static int[] memo = new int[14];
    static String[] pDiff = {"Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"};

    private static int monthDays(int m) {
        if (m < 1) {
            return 0;
        }
        if (memo[m] != 0) {
            return memo[m];
        }

        memo[m] = monthDays(m - 1) + mm[m - 1];
        return memo[m];
    }
    
    private static int numDays(int m, int d) {
        return (monthDays(m) + d);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int m1 = sc.nextInt();
        int d1 = sc.nextInt();
        int m2 = sc.nextInt();
        int d2 = sc.nextInt();

        int numDays1 = numDays(m1, d1);
        int numDays2 = numDays(m2, d2);

        int diff = numDays2 - numDays1;
        int remainder = diff % 7;

        while (remainder < 0) {
            remainder+= 7;
        }

        System.out.println(pDiff[remainder]);
    }
}

File: 240831/요일 맞추기/guess-day-of-week.kt
val inputList = readln().split(" ").map { it.toInt() }

val m1 = inputList[0]
val d1 = inputList[1]
val m2 = inputList[2]
val d2 = inputList[3]

val daysForMonth = intArrayOf(31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31)

fun main() {
    val past = (1 until m1).sumOf { month ->
        daysForMonth[month - 1]
    } + d1

    val now = (1 until m2).sumOf { month ->
        daysForMonth[month - 1]
    } + d2

    val diff = (now - past) % 7

    println(day(diff))
}

fun day(num: Int): String {
    when(num) {
        -6 -> return "Tue"
        -5 -> return "Wed"
        -4 -> return "Thu"
        -3 -> return "Fri"
        -2 -> return "Sat"
        -1 -> return "Sun"
        0 -> return "Mon"
        1 -> return "Tue"
        2 -> return "Wed"
        3 -> return "Thu"
        4 -> return "Fri"
        5 -> return "Sat"
        6 -> return "Sun"
        else -> error("invalid")
    }
}